docs/arch/controller.md

```diff
@@ -0,0 +1,40 @@
+# Architecture Document for Controller Component
+
+```mermaid
+sequenceDiagram
+    actor User
+    User->>Controller: sends pipeline jobs as YAML and repository link
+    alt is request malformed
+        Controller-->>User: Nok
+    else is well
+        Controller-->>User: Ok
+    end
+    Controller->>Database: saves pipeline in database
+    loop over steps
+        Controller-->>Scheduler: sends step (over GRPC)
+        Scheduler-->>Controller: job succeeded or not
```

Comment:
This seems really weird. You'll send "action" step by step ? I think you should send "action" by "action". Then other element will breakdown each action. 
If send step by step there is no point in defining actions imo. It will multiply network usage and keep the context of execution will be a nightmare.
Indeed, each action will need a context that the user will provide in the CI. Agent will receive "action" directly not steps.